src/Markup/Avalonia.Markup.Xaml/Data/Binding.cs

```diff
@@ -187,6 +191,69 @@ private static PathInfo ParsePath(string path)
                     result.ElementName = path.Substring(1);
                 }
             }
+            else if (path.StartsWith("$"))
+            {
+                var relativeSource = new RelativeSource();
+                result.RelativeSource = relativeSource;
+                var dot = path.IndexOf('.');
+                string relativeSourceMode;
+                if (dot != -1)
+                {
+                    result.Path = path.Substring(dot + 1);
+                    relativeSourceMode = path.Substring(1, dot - 1);
+                }
+                else
+                {
+                    result.Path = string.Empty;
+                    relativeSourceMode = path.Substring(1);
+                }
+
+                if (relativeSourceMode == "self")
+                {
+                    relativeSource.Mode = RelativeSourceMode.Self;
+                }
+                else if(relativeSourceMode == "parent")
+                {
+                    relativeSource.Mode = RelativeSourceMode.FindAncestor;
```

Comment:
This probably isn't what we want here, `RelativeSource` works on the visual tree, so:

```
<ContentControl>
  <TextBlock Text="$parent.Foo"/>
<ContentControl>
```

Will bind to the `ContentPresenter` in the `ContentControl` rather than the `ContentControl`. This is because we follow WPF's semantics for `RelativeSource` and that works on the visual rather than logical tree. This is OK in WPF because you have to specify the type of the ancestor, but here one would intuitively expect it to work on the logical tree.